tests/test_vault_api.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+
+import io
+import unittest
+import unittest.mock
+from urllib3.response import HTTPResponse
+
+from vault_cli import vault_python_api
+
+
+class TestVaultApi(unittest.TestCase):
+
+    def test_wrong_url(self):
+        """Check that an exception doesn't occur when URL provided by user is wrong"""
+        def send(self, request, stream=False, timeout=None, verify=True, cert=None, proxies=None):
+            data = b'404 page not found'
+            resp = HTTPResponse(body=io.BytesIO(data), preload_content=False)
+            resp.status = 404
+            resp.reason = 'Not Found'
+            resp.headers = {}
+            return self.build_response(request, resp)
+
+        with unittest.mock.patch('requests.adapters.HTTPAdapter.send', autospec=True, side_effect=send) as send:
+            # vault_python_api.url = 'http://127.0.0.1:8200/' # v1/ is missing
+            vault_python_api.userpass_authentication('user', 'pass')
+        self.assertEqual(send.call_count, 1)
```

Comment:
If we're starting to write tests, I'd rather have them with pytest :)

Reply:
I'll not start writing tests :)